### PR TITLE
update progressive_learner.py to fix the decider data splitting

### PR DESCRIPTION
**Pull request by latasianguy**
#### Reference issue

Closes issue #385

#### Type of change

Fixes bug with decider data splitting.

#### What does this implement/fix?

Previously, the decider is fitted with all data as it is not indexed by the `decider_idx`.
With this fix, it uses the amount of splitted data that it has access to.

#### Additional information

N/A

### CODE REVIEW

**Review by levinwil — APPROVED** [review]

**Review by PSSF23 — CHANGES_REQUESTED** [review]
@latasianguy Please `black` format the file you changed to pass travis.

Also I am a little confused here. @levinwil Based on my understanding, shouldn't the decider of a task have access to all data for that task?

**Comment by levinwil** [comment]
@PSSF23 Not necessarily. We might want the case that a decider only has access to a portion of the task data. For example, if we wanted to weight voter outputs based off of the accuracy of that voter, we would want to weight them based off of test accuracy. In that case, our "test set" (i.e. decider set) would need to be different than the voting set, since the voting set was used to train the voter - and therefore, testing on the voter set would overestimate generalization accuracy. 

I don't think doing any one thing with the dataset splits will always perform best. I think best split will change based on the dataset, so I think we should have the flexibility to specify the split, including decider sets that are subsets of the whole dataset.

**Comment by PSSF23** [comment]
@levinwil Okay, so `decider_idx` is not "the index of the decider" but "the indices of training data the decider has access to?"

**Review by PSSF23 — APPROVED** [review]

**Comment by levinwil** [comment]
@PSSF23 Correct: it's indices of the decider set.
